src/README.md

```diff
@@ -1,19 +1,101 @@
 
-See config.yaml for details of configuration options.
+### `netapp-storage-family`
+
+The storage family type that is used for the storage system. Can be "ontap_cluster" for clustering data ONTAP, or "eseries".
+
+### `netapp-storage-protocol`
+
+The SAN protocol to use. Can be either "iscsi" or "NFS".
+
+### `netapp-vserver`
+
+Specifies the virtual storage server name on the storage cluster.
+
+### `netapp-server-hostname`
+
+The hostname or IP address for the storage server (can alternatively specify a proxy server).
+
+### `netapp-server-port`
+
+The TCP port used to communicate with the storage server or proxy.
+
+If it's not specified, it will be deduced: For ONTAP drivers, it will be 80 for HTTP and 443 for HTTPS; for E-series, it will be 8080 and 8443, respectively.
+
+### `netapp-login`
+
+The username used to access the storage server or proxy.
+
+### `netapp-password`
+
+The password used to authenticate the "login" option.
+
+### `netapp-nfs-shares-config`
+
+Specifies a file that contains a list of NFS shares, each on its own line, to which the driver will attempt to provision
+Cinder volumes.
+
+### `netapp-controller-ips`
+
+If the storage family is "eseries", this option specifies a comma-separated list of controller hostnames or IP addresses
+to be used for provisioning.
+
+### `netapp-array-password`
+
+The password for the NetApp E-series storage array.
+
+### `netapp-storage-pools`
+
+Specifies a comma-separated list of pool names to use.
+
+### `use-multipath`
+
+Whether to use multipath for image transfer.
+
+### `netapp-enable-multiattach`
+
+Specifies whether the driver should allow operations that involve multiple attachments to a volume.
+
+### `volume-backend-name`
+
+The service name to present to Cinder.
+
+## Deployment
+
+This charm's primary use is as a backend for the cinder charm. To do so, add a relation betweeen both charms:
+
+  juju add-relation cinder-netapp:storage-backend cinder:storage-backend
+
+## Actions
+
+This section lists Juju [actions][juju-docs-actions] supported by the charm. Actions allow specific operations to be performed on a per-unit basis. To display action descriptions run `juju actions --schema cinder-netapp`. If the charm is not deployed then see file `actions.yaml`.
+
+* `pause`
+* `resume`
+* `restart-services`
+
+# Documentation
+
+The OpenStack Charms project maintains two documentation guides:
+
+* [OpenStack Charm Guide][cg]: for project information, including development
+  and support notes
+* [OpenStack Charms Deployment Guide][cdg]: for charm usage information
+
+# Bugs
+
+Please report bugs on [Launchpad][lp-bugs-charm-cinder-netapp].
+
+[cg]: https://docs.openstack.org/charm-guide
+[cdg]: https://docs.openstack.org/project-deploy-guide/charm-deployment-guide
+[lp-bugs-charm-cinder-netapp]: https://bugs.launchpad.net/charm-cinder-netapp/+filebug
```

Comment:
This doesn't seem to exist yet

Reply:
Yeah, I added this so that it's already in place for when the charm is incorporated into opendev.